source/transfer_list.rst

```diff
@@ -657,5 +660,116 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+.. _optee_pagable_part_entry:
+
+OPTEE pagable part entry layout (XFERLIST_OPTEE_PAGABLE_PART)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the address of OPTEE pagable part.
+This address is used when OPTEED is the Secure Payload Dispatcher.
+
+.. _tab_optee_pagable_part:
+.. list-table:: OPTEE pagable part type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **5**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of OPTEE pagable part in bytes.
+
+   * - pagable_part
+     - data_size
+     - hdr_size
+     - Pagable part address for OPTEED
+
+.. _tos_fw_config_entry:
+
+TOS FW config entry layout (XFERLIST_TOS_FW_CONFIG)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the address of SPMC manifest image.
+This address is used when OPTEE represents SPMC at S-EL1 and SPMD is the Secure Payload Dispatcher.
+
+.. _tab_tos_fw_config:
+.. list-table:: TOS FW config type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **6**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of TOS FW config in bytes.
+
+   * - tos_fw_config
+     - data_size
+     - hdr_size
+     - TOS FW config address for SPMD
+
+.. _opteed_aarch_entry:
+
+OPTEED AARCH entry layout (XFERLIST_OPTEED_AARCH)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the AARCH mode for OPTEED.
+This AARCH mode is used when OPTEED is the Secure Payload Dispatcher.
+
+.. _tab_opteed_aarch:
+.. list-table:: OPTEED AARCH type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **7**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of OPTEED AARCH mode in bytes.
+
+   * - opteed_aarch
+     - data_size
+     - hdr_size
+     - Holds the OPTEED AARCH mode.
```

Comment:
You should clarify what exactly is stored here (e.g. is it an enum value? Is it a literal `32` or `64` integer?).

Assuming this is just aarch32 vs aarch64(?) and can thus be represented as one bit, you may want to consider defining it as a bit field so that the other values are available for future expansion.

Reply:
I will prefer to use 4 bytes here if there are no special requirements, as right now OPTEED uses a type of 'uint32_t' to represent aarch mode during setup, in this case, we don't need to interpret the TE data after we parse from the TL.

Reply:
But as Julius says, you do need to define the values

Reply:
Definitions added.

Reply:
I would recommend to put those definitions down here in the entry for this specific field, not at the top of the TE (for TEs with one field it doesn't make a ton of difference, but since we may also get larger TEs with more fields it would make sense to standardize on keeping everything with the field it belongs to).

Reply:
Resolve this since the TE for AARCH is removed.